plugins/instanceof-checkcast/src/main/java/org/qbicc/plugin/instanceofcheckcast/LowerClassInitCheckBlockBuilder.java

```diff
@@ -31,26 +32,29 @@ public LowerClassInitCheckBlockBuilder(final CompilationContext ctxt, final Basi
         this.ctxt = ctxt;
     }
  
-    public Node initCheck(final ObjectType objectType) {
-        doInitCheck(objectType);
-        return nop();
-    }
+    public Node initCheck(InitializerElement initializer) {
+        if (initializer.hasAllModifiersOf(ClassFile.I_ACC_RUN_TIME)) {
+            // generate a run time init check
+
+            // and call the initializer
+            call(initializerOf(initializer), List.of());
```

Comment:
The approach looks good.  Is the part that limits the initializer to only being executed once with this approach still TBD?